SimMuon/GEMDigitizer/src/GEMDigiProducer.cc

```diff
@@ -27,7 +27,8 @@ namespace CLHEP {
 }
 
 GEMDigiProducer::GEMDigiProducer(const edm::ParameterSet& ps)
-  : digiModelString_(ps.getParameter<std::string>("digiModelString"))
+  : gemDigiModel_{GEMDigiModelFactory::get()->create("GEM" + ps.getParameter<std::string>("digiModelString") + "Model",
+                                                     ps)}
 {
```

Comment:
@makortel , do you not think that complex initialisation within constructor may be difficult for code reader? I understand that this code now is more effective than previous variant, however, may be enough simple removal of digiModelString_ class member?

Reply:
I don't think replacing `digiModelString_` with `ps.getParameter<std::string>("digiModelString")` would make the code much harder to read. Although my main motivation for this construct was to avoid additional `std::unique_ptr<GEMDigiModel>{GEMDigiModelFactory::get()...}` that would be needed for the assignment in the constructor body.